src/util/object.js

```diff
@@ -40,5 +40,16 @@ export function copyObject(object) {
 // compare objects with stringify
 // works for everything except circular refs, functions, and js Dates
 export function compareObjects(object1, object2) {
-  return JSON.stringify(object1) === JSON.stringify(object2);
+  if (typeof object1 !== typeof object2)
```

Comment:
Does the new approach have better performance because of the added shortcut comparisons?

Reply:
It should be a little better I think, with the short-circuits, yes.

Though it won’t matter as much because I got rid of as many calls to this function as I could.

Reply:
It should be a little better I think, with the short-circuits, yes.

Though it won’t matter as much because I got rid of as many calls to this function as I could.